src/transformers/models/swiftformer/modeling_tf_swiftformer.py

```diff
@@ -0,0 +1,911 @@
+            tf.keras.layers.Dense(self.num_labels, name="dist_head")
+            if self.num_labels > 0
+            else tf.keras.layers.Identity(name="dist_head")
+        )
+
+    @unpack_inputs
+    @add_start_docstrings_to_model_forward(TFSWIFTFORMER_INPUTS_DOCSTRING)
+    def call(
+        self,
+        pixel_values: Optional[tf.Tensor] = None,
+        labels: Optional[tf.Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        training: bool = False,
+    ) -> Union[tuple, TFImageClassifierOutputWithNoAttention]:
+        r"""
+        labels (`tf.Tensor` of shape `(batch_size,)`, *optional*):
+            Labels for computing the image classification/regression loss. Indices should be in `[0, ...,
+            config.num_labels - 1]`. If `config.num_labels == 1` a regression loss is computed (Mean-Square loss), If
+            `config.num_labels > 1` a classification loss is computed (Cross-Entropy).
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        # run base model
+        outputs = self.swiftformer(
+            pixel_values,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            training=training,
+        )
+
+        sequence_output = outputs.last_hidden_state if return_dict else outputs[0]
+        sequence_output = tf.transpose(sequence_output, perm=[0, 2, 3, 1])
+
+        # run classification head
+        sequence_output = self.norm(sequence_output, training=training)
+        # FIXME: review this conversion
+        sequence_output = tf.transpose(sequence_output, perm=[0, 3, 1, 2])
+        _, c, h, w = sequence_output.shape
+        sequence_output = tf.reshape(sequence_output, [-1, c, h * w])
+        sequence_output = tf.reduce_mean(sequence_output, axis=-1)
+        cls_out = self.head(sequence_output)
+        distillation_out = self.dist_head(sequence_output)
+        logits = (cls_out + distillation_out) / 2
+
+        # calculate loss
+        loss = None
+        if labels is not None:
+            if self.config.problem_type is None:
+                if self.num_labels == 1:
+                    self.config.problem_type = "regression"
+                elif self.num_labels > 1 and (
+                    labels.dtype == tf.int64 or labels.dtype == tf.int32
+                ):  # FIXME: is this it?
+                    self.config.problem_type = "single_label_classification"
+                else:
+                    self.config.problem_type = "multi_label_classification"
+
+            if self.config.problem_type == "regression":
+                loss_fct = tf.keras.losses.MSE
+                if self.num_labels == 1:
+                    loss = loss_fct(labels.squeeze(), logits.squeeze())
+                else:
+                    loss = loss_fct(labels, logits)
+            # FIXME: multilabel or multiclass?
```

Comment:
There's a whole bunch of hanging fixme's here

Reply:
Removed most of them as I checked them